packages/flutter/test/widgets/focus_traversal_test.dart

```diff
@@ -1874,6 +1875,113 @@ void main() {
       expect(controller.offset, equals(0.0));
     }, skip: isBrowser, variant: KeySimulatorTransitModeVariant.all()); // https://github.com/flutter/flutter/issues/35347
 
+    testWidgets('Focus traversal inside a horizontal scrollable applies ensure visible padding.', (WidgetTester tester) async {
```

Comment:
Can you add a test for a vertical scroll view? Also one that is reversed, those are often not considered.